tree_in_context/src/typechecker.rs

```diff
@@ -51,19 +51,25 @@ impl Expr {
     }
 }
 
+/// A map from the old, untyped expression to the new, typed expression
+pub type TypedExprCache = HashMap<*const Expr, RcExpr>;
```

Comment:
what's the difference between TypeExprCache and TypeCache?

Reply:
Good question, I wrote new documentation to try to clarify
```
/// Typechecking produces new, typed expressions.
/// This map is used to memoize the results of typechecking.
/// It maps the old untyped expression to the new typed expression
pub type TypedExprCache = HashMap<*const Expr, RcExpr>;

/// We also need to keep track of the type of the newly typed expression.
/// This maps the newly instrumented expression to its type.
pub type TypeCache = HashMap<*const Expr, Type>;
```